ntbs-service/DataAccess/DataQualityRepository.cs

```diff
@@ -297,32 +322,34 @@ public class NotificationAndDuplicateIds
             public int DuplicateId { get; set; }
         }
 
-        private IQueryable<Notification> GetBaseNotificationQueryableForAlerts()
+        private IQueryable<Notification> GetBaseNotificationQueryableForAlerts(AlertType alertType)
         {
             return _context.Notification
                 .Include(n => n.HospitalDetails)
+                .Include(n => n.Alerts)
                 .Where(n => n.NotificationStatus != NotificationStatus.Closed
                             && n.NotificationStatus != NotificationStatus.Deleted
-                            && n.NotificationStatus != NotificationStatus.Denotified);
+                            && n.NotificationStatus != NotificationStatus.Denotified
+                            && n.Alerts.All(a => a.AlertType != alertType));
```

Comment:
Just want to make sure i understand what the intention behind this change is: 
* we are trying to get notifications which are eligable for alert of type `alertType`
* ... and that means they can't already have this alert type on them
* ... so we filter out any notification that already has an alert of this type
is that correct?

If so, I think that's quite a neat trick! But it probably deserves a comment on the method, as it is very much not what you would expect from reading the signature of the function!

Also I would prefer to use the type, rather than the `AlertType` enum - I find the enum quite redundant and messy. Could we do 
```
private IQueryable<Notification> GetBaseNotificationQueryableForAlerts<T>() where T : Alert { 
  ...
  .Where ( ... n.Alerts.All(a => a is T))
 ...
}
```

Reply:
yeap, that's correct. I have changes name of this private method and added comments